### PR TITLE
ARROW-1719: [Java] Remove accessor and mutator interface

### PR DESCRIPTION
**Pull request by icexelloss**

### CODE REVIEW

**Comment by icexelloss** [comment]
cc @BryanCutler

File: java/vector/src/test/java/org/apache/arrow/vector/TestVectorReAlloc.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ public void testMapType() {
       assertEquals(512, vector.getValueCapacity());
 
       try {
-        vector.getAccessor().getObject(513);
+        vector.getObject(513);
```
Comment (BryanCutler) [inline java/vector/src/test/java/org/apache/arrow/vector/TestVectorReAlloc.java]: Wow, this test should not have been passing...

**Review by BryanCutler — APPROVED** [review]
+1, looks great!

**Comment by icexelloss** [comment]
cc @siddharthteotia.

Can you also take a look please? Thanks!

**Comment by icexelloss** [comment]
Thanks @BryanCutler for the review!

**Comment by siddharthteotia** [comment]
+1

**Pull request closed without merging**